### PR TITLE
SONAR-8288 WS api/measures/search_projects DSL handle isFavorite

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java [hunk 1]
```diff
@@ -57,11 +57,15 @@
   private final DbClient dbClient;
   private final ProjectMeasuresIndex index;
   private final ProjectMeasuresQueryValidator projectMeasuresQueryValidator;
+  private final ProjectMeasuresQueryFactory queryFactory;
+  private final UserSession userSession;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java]: MAJOR Remove this unused "userSession" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java [hunk 2]
```diff
@@ -57,11 +57,15 @@
...
 
-  public SearchProjectsAction(DbClient dbClient, ProjectMeasuresIndex index, ProjectMeasuresQueryValidator projectMeasuresQueryValidator) {
+  public SearchProjectsAction(DbClient dbClient, ProjectMeasuresIndex index, ProjectMeasuresQueryFactory queryFactory, ProjectMeasuresQueryValidator projectMeasuresQueryValidator, UserSession userSession) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java]: MINOR Split this 206 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/component/es/ProjectMeasuresQuery.java [hunk 3]
```diff
@@ -56,6 +57,19 @@ public boolean hasQualityGateStatus() {
...
+    this.projectUuids = requireNonNull(projectUuids);
+    return this;
+  }
+
+  public boolean doesFilterOnProjectUuids() {
+    return projectUuids != null;
+  }
+
+  public List<String> getProjectUuids() {
+    return requireNonNull(projectUuids);
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/es/ProjectMeasuresQuery.java]: I would prefer to return an IllegalStateException in this case

**Pull request merged**